.github/workflows/main.yaml

```diff
@@ -286,6 +310,41 @@ jobs:
           integration-enabled: 'false'
           deployed-via-kubectl: 'true'
 
+      - name: Upgrade via kubectl as subchart in production mode
+        run: |
+          oldpodname=$(kubectl get pods -l app.kubernetes.io/name=replicated -o jsonpath='{.items[0].metadata.name}')
+
+          helm template test-chart oci://registry.replicated.com/$APP_SLUG/$CHANNEL_SLUG/test-chart --set replicated.integration.enabled=false --set replicated.versionLabel=1.0.0 | kubectl apply -f -
+          kubectl rollout status deployment test-chart --timeout=2m
+          kubectl rollout status deployment replicated --timeout=2m
+
+          COUNTER=1
+          while [ kubectl get pods -l app.kubernetes.io/name=replicated -o jsonpath='{.items[0].metadata.name}' | grep -q $oldpodname ]; do
+            ((COUNTER += 1))
+            if [ $COUNTER -gt 60 ]; then
+              echo "Pod did not restart after upgrade"
+              exit 1
+            fi
+            sleep 1
+          done
+
+          newpodname=$(kubectl get pods -l app.kubernetes.io/name=replicated -o jsonpath='{.items[0].metadata.name}')
```

Comment:
Why doesn't this logic exist for the other upgrade test scenario?

Reply:
this shouldn't be there since we check the pods above. still WIP on the testing front, so will request another review once it's ready.